### PR TITLE
feat: working with iframes

### PR DESCRIPTION
**Pull request by duvany10**
working with iframes

### CODE REVIEW

File: cypress/page/iframe.page.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    private w3schoolSiteUrl: string;
+    private defaulIframe: string;
+    private cssH1: string;
+    private tutorialSpan: string;
+
+    constructor() {
+        this.w3schoolSiteUrl = 'https://www.w3schools.com/html/html_iframe.asp';
+        this.defaulIframe = '[src="default.asp"]';
+        this.cssH1 = '#main > h1';
+        this.tutorialSpan = '#main > h1 > span';
```
Comment (holgiosalos) [inline cypress/page/iframe.page.ts:12]: this property is not required

File: cypress/page/iframe.page.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    private w3schoolSiteUrl: string;
+    private defaulIframe: string;
+    private cssH1: string;
+    private tutorialSpan: string;
+
+    constructor() {
+        this.w3schoolSiteUrl = 'https://www.w3schools.com/html/html_iframe.asp';
+        this.defaulIframe = '[src="default.asp"]';
+        this.cssH1 = '#main > h1';
```
Comment (holgiosalos) [inline cypress/page/iframe.page.ts:11]: if this selector is not used, then remove it

File: cypress/page/iframe.page.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        return cy.iframe(this.defaulIframe);
+    }
+
+    public goToCssPageInFrame() {
+        cy.iframe(this.defaulIframe).find('a[title="CSS Tutorial"]').click();
+    }
+
+    public getCSSh1Title() {
+        return cy.iframe(this.defaulIframe);
+    }
```
Comment (holgiosalos) [inline cypress/page/iframe.page.ts:29]: as this method is the same than getFrameTitle you can remove this method and use `getFrameTitle` for both tests

File: cypress/page/iframe.page.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    public visit(): void {
+        cy.visit(this.w3schoolSiteUrl);
+    }
+
+    public getFrameTitle() {
+        return cy.iframe(this.defaulIframe);
+    }
+
+    public goToCssPageInFrame() {
+        cy.iframe(this.defaulIframe).find('a[title="CSS Tutorial"]').click();
```
Comment (holgiosalos) [inline cypress/page/iframe.page.ts:24]: the text used in find should be a selector and should be a class property

File: cypress/integration/iframe.spec.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      const CSSTutorialExpected = "CSS Tutorial";
+
+      // Act
+      iframePage.visit();
+
+      // Assert
+      iframePage.getFrameTitle().should("contain.text", HTMLTutorialExpected);
+      iframePage.goToCssPageInFrame();
+      iframePage.getFrameTitle().should("contain.text", CSSTutorialExpected);
+    });
```
Comment (holgiosalos) [inline cypress/integration/iframe.spec.ts:22]: - are expected two separated tests, you can create a before hook and put the call to the visit method there (which is not the act, it is part of the arrange)
- first test, should have the call to the getFrameTitle with the assertion of the HTML tutorial.
- The second test should have the call to the goToCssPageInFrame and the call again to the getFrameTitle with the assertion of the CSS tutorial.

File: cypress/page/iframe.page.ts (hunk 1 above)
Comment (duvany10) [reply cypress/page/iframe.page.ts:12]: Done!

File: cypress/page/iframe.page.ts (hunk 2 above)
Comment (duvany10) [reply cypress/page/iframe.page.ts:11]: Done!

File: cypress/page/iframe.page.ts (hunk 3 above)
Comment (duvany10) [reply cypress/page/iframe.page.ts:29]: Done!

File: cypress/page/iframe.page.ts (hunk 4 above)
Comment (duvany10) [reply cypress/page/iframe.page.ts:24]: Done!

**Review by holgiosalos — APPROVED** [review]

**Comment by duvany10** [comment]
Thank you @holgiosalos

**Pull request merged**